### ISSUE TITLE
gitfiles misbehaving

### ISSUE THREAD

**Issue by pfuntner** [pfuntner/toys#64]
I was working with a small new repo for a gist and had trouble using `gitfiles`:

```
$ git remote -v
origin  git@gist.github.com:19d6e0abbea44001c5d1c4e409e29dd8.git (fetch)
origin  git@gist.github.com:19d6e0abbea44001c5d1c4e409e29dd8.git (push)
$ gitfiles
Traceback (most recent call last):
  File "/home/centos/bin/gitfiles", line 146, in <module>
    report(brief=True)
  File "/home/centos/bin/gitfiles", line 48, in report
    assert date and text, "At line %d, previous commit (%s) was not closed" % (lineno, commit)
AssertionError: At line 5, previous commit (5946a2e8248b46e9e5285121a394504a26c355de) was not closed
$ git log --name-only
commit 5946a2e8248b46e9e5285121a394504a26c355de
Author: John Pfuntner <pfuntner@pobox.com>
Date:   Tue Jan 14 19:36:37 2020 +0000

packer-ssh-handshake-hang

commit 40904a8a0187ca9aabe00cdc8ec79d672bb6003a
Author: John Pfuntner <jpfuntne@cisco.com>
Date:   Tue Jan 14 14:27:04 2020 -0500

    Committing files to support issue

README.md
aws-ebs-ansible.json
debug.log
harden.yml
vars-rhel8.json

commit 3dcf422cd8aaf78405d2f051bf2aa2ea8951827e
Author: John Pfuntner <pfuntner@pobox.com>
Date:   Tue Jan 14 19:16:46 2020 +0000

packer-ssh-handshake-hang
$
```

I don't know if this is a Python 3 issue or not.

**Comment by pfuntner** [comment]
I'm not sure what precisely is wrong.  It may have to do with the first commit having no comment lines.  Regardless:

- `git-log` also seems to have the same problem
- my `Git` class in my `gityup.py` script is able to parse things better
- I think I should remove the parsing from `git-log` and `git-files` and replace it with a call to my my `Git` class.

**Comment by pfuntner** [comment]
Fixed in 9af2fb8ad3aab8c248c63b301283be48e49e611a.  I also did a lot of the same work in `git-log` in 91d45ac86d661fea41acc4b8aafe7ec10dd33c9b
